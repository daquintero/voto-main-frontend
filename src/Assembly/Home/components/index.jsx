// General Home CorruptionCase Index
import React from 'react';
// import PropTypes from 'prop-types';
import {
  Container,
  Row,
  Col,
} from 'reactstrap';

const Home = () => (
  <Container>
    <Row>
      <Col>
        gg
      </Col>
    </Row>
  </Container>
);

export default Home;
