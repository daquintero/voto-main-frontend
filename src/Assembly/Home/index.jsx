// Home Page According to their wishes
// Libraries
import React, { PureComponent } from 'react';
import { Container, Row, Col } from 'reactstrap';

// Components
import Video from '../Video';
import Commit from './components/Commit';
import PDF from './components/pdfPreviewer';

// Declaration
class Home extends PureComponent {
  render() {
    return (
      <>
        <Container fluid className="p-0">
          <Video />
        </Container>
        <Commit light />
        <Container>
          <Row noGutters>
            <Col md={6}>
              asdf
            </Col>
            <Col md={6} className="text-center">
              <a href="https://s3.amazonaws.com/votoinformado2019/Manual_Voto_Informado_2019.pdf">
                <h4 className="p-2 m-0">
                  <i className="fal fa-book-heart p-2" />
                  Â¡Descarga el Manual!
                </h4>
              </a>
              <PDF />
            </Col>
          </Row>
        </Container>
      </>
    );
  }
}

// TODO State Store Connection
export default Home;
