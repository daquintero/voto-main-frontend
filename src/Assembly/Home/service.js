import axios from 'axios';
import buildUrl from '../../shared/utils/buildUrl';


const baseUrl = process.env.REACT_APP_BASE_URL;
const homeApiUrl = `${baseUrl}/home/api/v1/`;
const searchApiUrl = `${baseUrl}/search/api/v1/`;
const subscribedStatsApiUrl = `${baseUrl}/home/api/v1/subscribedStats/`;
const discoverCandidatesApiUrl = `${baseUrl}/home/api/v1/discoverCandidates/`;

const urls = {
  get: {
    home: homeApiUrl,
    subscribedStats: subscribedStatsApiUrl,
    discoverCandidates: discoverCandidatesApiUrl,
  },
  search: {
    home: searchApiUrl,
  },
};


// GET
const getHome = () => axios.get(urls.get.home);
const subscribedStats = () => axios.get(urls.get.subscribedStats);
const discoverCandidates = gid => axios.get(buildUrl(urls.get.discoverCandidates, { gid }));

// Search
const homeSearch = searchData => axios.get(buildUrl(urls.search.home, searchData));

const service = {
  get: {
    home: getHome,
    subscribedStats,
    discoverCandidates,
  },
  search: {
    home: homeSearch,
  },
};

export default service;
