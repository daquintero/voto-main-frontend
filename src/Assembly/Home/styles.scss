.home {
  &__wrapper {
    background-color: $colorBackgroundBody;
  }
}
