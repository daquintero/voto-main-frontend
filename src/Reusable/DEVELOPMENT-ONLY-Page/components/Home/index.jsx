// Social Page
// Libraries
import React, { PureComponent } from 'react';
import { Col, Container, Row } from 'reactstrap';
import { Link } from 'react-router-dom';

// Prop Types
import PropTypes from 'prop-types';

// Components
import MainGrid from './components/Main';
import Search from './components/Search';

const banner = `${process.env.PUBLIC_URL}/img/banner.svg`;

// Declaration
class Whatever extends PureComponent {
  static propTypes = {
    mainResults: PropTypes.arrayOf(Object).isRequired,
    searchResults: PropTypes.arrayOf(Object).isRequired,
  };
  render() {
    const { mainResults, searchResults } = this.props;
    return (
      <>
        <Container fluid className="p-0">
          <Row noGutters className="py-4 bg-light">
            <Col sm={6} md={4} className="px-4 py-4 py-sm-0">
              <img src={banner} alt="Error" />
            </Col>
            <Col sm={6} md={8} className="px-4">
              <h3>¡Bienvenida(o) a #VotoInformado2019!</h3>
              <p className="pr-5 m-0">
                Somos voluntarios sin relación a partidos
                políticos, organizaciones políticas ni candidatos
                que buscamos rescatar y revitalizar el voto como instrumento de
                cambio y progreso en nuestra sociedad.
              </p>
              <Link to="/acerca">
                Averigua más
              </Link>
            </Col>
          </Row>
        </Container>
        <Container>
          <MainGrid instances={mainResults} />
          <Search instances={searchResults} />
        </Container>
      </>
    );
  }
}

// TODO State Store Connection
export default Whatever;
