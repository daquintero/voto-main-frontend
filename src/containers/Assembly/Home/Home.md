## Home Page

The youtube video is embedded using
 responsive 100% design for autoplay
 upon arrival to the site.
