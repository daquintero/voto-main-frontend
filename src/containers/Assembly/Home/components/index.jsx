// General Home Page Index
// TODO input might be moved to menu
import React from 'react';
// import PropTypes from 'prop-types';
import {
  Container,
  Row,
  Col,
  FormGroup,
  Input,
} from 'reactstrap';

const Home = () => (
  <Container>
    <Row>
      <Col>
        <FormGroup>
          <Input
            type="string"
            name="text"
            id="homeSearch"
            placeholder="Investiga!"
          />
        </FormGroup>
      </Col>
    </Row>
  </Container>
);

export default Home;
