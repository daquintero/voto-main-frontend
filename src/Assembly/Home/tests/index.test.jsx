// Test for the Home CorruptionCase Component
import React from 'react';
import renderer from 'react-test-renderer';
import Home from '..';

describe('<Home />', () => {
  it('HomePage renders correctly without crashing', () => {
    const tree = renderer
      .create(<Home />)
      .toJSON();
    expect(tree)
      .toMatchSnapshot();
  });
});
