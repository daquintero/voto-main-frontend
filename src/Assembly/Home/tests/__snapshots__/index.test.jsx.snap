// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> HomePage renders correctly without crashing 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      gg
    </div>
  </div>
</div>
`;
